src/ui-kit/experimental/hierarchical/hierarchical-interface.ts

```diff
@@ -14,7 +14,7 @@ export interface SamHiercarchicalServiceInterface {
      * 
      * @param id 
      */
-    getHiercarchicalById(id?: string, searchValue?: string): Observable<object[]>;
+    getHiercarchicalById(id: string, searchValue: string, sort: Sort): Observable<object[]>;
```

Comment:
@thomastighe Just a note that this makes a breaking change since the params are no longer optional. Just make sure we do a "minor" update over a patch next publish

Reply:
I will note that. I think I had the ? because I was being lazy. I am debating turning that an object since it will have a 4th param shortly. Would like your opinion on that.

Reply:
@thomastighe Crockford would say pass the args as a single object. I'm not sure its always worth the effort, its more stuff for your consumers to learn and more layers of abstraction to sift through. 

In this case, with 3 params and a fourth soon to come, plus the fact that we don't want to have to change the internals in the event this service interface needs modification, an object is probably a better idea.

If you can use the builder pattern to generate the config on this, even better.